### ISSUE TITLE
create-test / default-params issue

### ISSUE THREAD

**Issue by xjamundx** [eslint/espree#55]
1. the placeholder for empty defaults array members should possibly be `undefined` instead of `null`
2. `create-test-.js` improperly changes `undefined` array members to `null`.

Recently when working on `defaultParams` and `destructuring` I noticed that I was having some mismatches with Esprima on the expected output, especially involving the placeholder when there are not defaults specified for each parameter.

When I use `create-test` to generate the expected output it seems to show `[null, 1]` for `function(a, b=1){}`, but when i actually `console.log` the output from Esprima, it looks like `[undefined, 1]`.

Here is I think the source of the issue:
![screen shot 2015-02-09 at 7 00 23 am](https://cloud.githubusercontent.com/assets/246143/6108656/c8c33dda-b029-11e4-9ee0-79f1208e33be.png)

The only thing that makes me unsure is that the single test in esprima also shows `null` as the placeholder, but I can only assume they're using JSON.stringify in their tests, because Esprima will not generate that output:
https://github.com/jquery/esprima/blob/harmony/test/harmonytest.js#L2612

Let me know if you have any ideas otherwise I'll like manually change the tests to use `undefined` as part of #51, which changed some of the code around this area and caused me to bump into this issue.

**Comment by nzakas** [comment]
Ah, interesting. Was this causing an error during testing? Since null == undefined, it would seem like a semantic difference but not a functional one.

**Comment by xjamundx** [comment]
When I updated to the latest eprima version of `parseParam` method for the `destructuring` feature, the old `defaultParams` tests started failing. The harmony version looks like this:
https://github.com/ariya/esprima/blob/harmony/esprima.js#L4586
```
options.defaults.push(def);
```

In order to not break tests I had to:
https://github.com/eslint/espree/pull/51/files#diff-7a476a3203e745831344f37dd24a8a97R4293

```
options.defaults.push(def ? def : null);
```

And yes I checked everywhere upstream and made sure there was no setting of `null` anywhere else. I actually have a hunch that there used to be some sort of explicit `null`ing happening, but that it was removed and for whatever reason there tests still passed, whereas ours are stricter or something.

Kind of weird thing, I'll go with whatever is the path of least resistance for now, which will be to leave the `def ? def : null` fix in there and then maybe I can check with esprima and figure out what is the official approach.

**Comment by xjamundx** [comment]
Probably it doesn't really matter, because it's such an edge case, but if we want to fix the `create-test` tool we may be able to use `util.inspect` instead of `JSON.stringify`. Here is what the difference is like:

```js
var x = {
    a: [undefined, undefined],
    b: {x: undefined }
};

util.inspect(x, {depth: null}) // { a: [ undefined, undefined ], b: { x: undefined } }

JSON.stringify(x); // {"a":[null,null],"b":{}}
```

The only issue I'm seeing is that I'm not sure how to format `util.inspect` for multiple lines.

**Comment by xjamundx** [comment]
Realizing this is a super small edge case and not likely worth fixing, especially as the way default params are being handled is changing anyway https://github.com/jquery/esprima/issues/1081
